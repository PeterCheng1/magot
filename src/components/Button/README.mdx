---
name: Button 按钮
route: /components/Button
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Button from './Button';

# Button 按钮

按钮组件，用于响应用户的点击交互行为，触发特定的业务逻辑。

## Usage

### 按钮类型

<Playground>
  <Button>Default</Button>
  <Button type="primary">Primary</Button>
  <Button type="success">Success</Button>
  <Button type="warning">Warning</Button>
  <Button type="danger">Danger</Button>
  <Button type="text">Text</Button>
  <Button type="link">Link</Button>
</Playground>

### 按钮尺寸

<Playground>
  <div className="demo-group">
    <Button size="small">Small</Button>
    <Button>Normal</Button>
    <Button size="large">Large</Button>
  </div>
  <div className="demo-group">
    <Button type="primary" size="small">
      Small
    </Button>
    <Button type="primary">Normal</Button>
    <Button type="primary" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="success" size="small">
      Small
    </Button>
    <Button type="success">Normal</Button>
    <Button type="success" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="warning" size="small">
      Small
    </Button>
    <Button type="warning">Normal</Button>
    <Button type="warning" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="danger" size="small">
      Small
    </Button>
    <Button type="danger">Normal</Button>
    <Button type="danger" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="text" size="small">
      Small
    </Button>
    <Button type="text">Normal</Button>
    <Button type="text" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="link" size="small">
      Small
    </Button>
    <Button type="link">Normal</Button>
    <Button type="link" size="large">
      Large
    </Button>
  </div>
</Playground>

### 不可用状态

<Playground>
  <div className="demo-group">
    <Button>Default</Button>
    <Button disabled={true}>Default(disabled)</Button>
  </div>
  <div className="demo-group">
    <Button type="primary">Primary</Button>
    <Button type="primary" disabled={true}>
      Primary(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="success">Success</Button>
    <Button type="success" disabled={true}>
      Success(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="warning">Warning</Button>
    <Button type="warning" disabled={true}>
      Warning(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="danger">Danger</Button>
    <Button type="danger" disabled={true}>
      Danger(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="text">Text</Button>
    <Button type="text" disabled={true}>
      Text(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="link" href="https://github.com/billjs/magot" target="_blank">
      Link
    </Button>
    <Button
      type="link"
      href="https://github.com/billjs/magot"
      target="_blank"
      disabled={true}>
      Link(disabled)
    </Button>
  </div>
</Playground>

### 图标按钮

<Playground>
  <div className="demo-group">
    <Button size="small" icon="download">
      Default
    </Button>
    <Button type="primary" size="small" icon="download">
      Primary
    </Button>
    <Button type="success" size="small" icon="smile">
      Success
    </Button>
    <Button type="warning" size="small" icon="return">
      Warning
    </Button>
    <Button type="danger" size="small" icon="delete">
      Danger
    </Button>
    <Button type="text" size="small" icon="comment">
      Text
    </Button>
    <Button type="link" size="small" icon="message">
      Link
    </Button>
  </div>
  <div className="demo-group">
    <Button icon="download">Default</Button>
    <Button type="primary" icon="download">
      Primary
    </Button>
    <Button type="success" icon="smile">
      Success
    </Button>
    <Button type="warning" icon="return">
      Warning
    </Button>
    <Button type="danger" icon="delete">
      Danger
    </Button>
    <Button type="text" icon="comment">
      Text
    </Button>
    <Button type="link" icon="message">
      Link
    </Button>
  </div>
  <div className="demo-group">
    <Button icon="download" size="large">
      Default
    </Button>
    <Button type="primary" size="large" icon="download">
      Primary
    </Button>
    <Button type="success" size="large" icon="smile">
      Success
    </Button>
    <Button type="warning" size="large" icon="return">
      Warning
    </Button>
    <Button type="danger" size="large" icon="delete">
      Danger
    </Button>
    <Button type="text" size="large" icon="comment">
      Text
    </Button>
    <Button type="link" size="large" icon="message">
      Link
    </Button>
  </div>
  <div className="demo-group">
    <Button disabled={true} icon="download">
      Default
    </Button>
    <Button type="primary" disabled={true} icon="download">
      Primary
    </Button>
    <Button type="success" disabled={true} icon="smile">
      Success
    </Button>
    <Button type="warning" disabled={true} icon="return">
      Warning
    </Button>
    <Button type="danger" disabled={true} icon="delete">
      Danger
    </Button>
    <Button type="text" disabled={true} icon="comment">
      Text
    </Button>
    <Button type="link" disabled={true} icon="message">
      Link
    </Button>
  </div>
</Playground>

#### 图标在右侧

<Playground>
  <Button iconPosition="right" icon="download">
    Default
  </Button>
  <Button type="primary" iconPosition="right" icon="download">
    Primary
  </Button>
  <Button type="success" iconPosition="right" icon="smile">
    Success
  </Button>
  <Button type="warning" iconPosition="right" icon="return">
    Warning
  </Button>
  <Button type="danger" iconPosition="right" icon="delete">
    Danger
  </Button>
  <Button type="text" iconPosition="right" icon="comment">
    Text
  </Button>
  <Button type="link" iconPosition="right" icon="message">
    Link
  </Button>
</Playground>

#### 纯图标按钮

<Playground>
  <Button icon="download" />
  <Button type="primary" icon="download" />
  <Button type="success" icon="smile" />
  <Button type="warning" icon="return" />
  <Button type="danger" icon="delete" />
  <Button type="text" icon="comment" />
  <Button type="link" icon="message" />
</Playground>

#### 圆角纯图标按钮

<Playground>
  <Button icon="scan" circular={true} size="small" />
  <Button icon="scan" circular={true} />
  <Button icon="scan" circular={true} size="large" />
  <Button type="primary" icon="download" circular={true} size="small" />
  <Button type="primary" icon="download" circular={true} />
  <Button type="primary" icon="download" circular={true} size="large" />
  <Button type="danger" icon="delete" circular={true} size="small" />
  <Button type="danger" icon="delete" circular={true} />
  <Button type="danger" icon="delete" circular={true} size="large" />
</Playground>

#### 方形纯图标按钮

<Playground>
  <Button icon="scan" square={true} size="small" />
  <Button icon="scan" square={true} />
  <Button icon="scan" square={true} size="large" />
  <Button type="primary" icon="download" square={true} size="small" />
  <Button type="primary" icon="download" square={true} />
  <Button type="primary" icon="download" square={true} size="large" />
  <Button type="danger" icon="delete" square={true} size="small" />
  <Button type="danger" icon="delete" square={true} />
  <Button type="danger" icon="delete" square={true} size="large" />
</Playground>

#### 加载中按钮

<Playground>
  <div className="demo-group">
    <Button loading={true}>Loading</Button>
    <Button type="primary" loading={true}>
      Loading
    </Button>
    <Button type="text" loading={true}>
      Loading
    </Button>
    <Button loading={true} circular={true} />
    <Button type="primary" loading={true} circular={true} />
  </div>
</Playground>

### Block 按钮

<Playground>
  <Button block={true}>Default</Button>
  <Button type="primary" block={true}>
    Primary
  </Button>
  <Button type="success" block={true}>
    Success
  </Button>
  <Button type="warning" block={true}>
    Warning
  </Button>
  <Button type="danger" block={true}>
    Danger
  </Button>
  <Button type="text" block={true}>
    Text
  </Button>
  <Button type="link" block={true}>
    Link
  </Button>
</Playground>

### 按钮点击交互

<Playground>
  <Button onClick={() => alert('You clicked the Button')}>Click me</Button>
  <Button type="primary" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="success" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="warning" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="danger" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="text" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="link" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
</Playground>

### Link 按钮

<Playground>
  <Button href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="primary" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="success" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="warning" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="danger" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="text" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="link" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
</Playground>

### 自定义样式按钮

<Playground>
  <Button
    style={{
      color: '#52c41a',
      backgroundColor: '#343a40',
      borderColor: '#343a40',
    }}>
    My Button
  </Button>
  <Button
    style={{
      color: '#fff',
      backgroundColor: '#f5222d',
      borderColor: '#f5222d',
      borderRadius: 0,
    }}>
    My Button
  </Button>
</Playground>

### 按钮组合

文档详见：[按钮组合](/components/ButtonGroup)。

<Playground>
  <Button.Group>
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
  <Button.Group>
    <Button type="primary" icon="arrow-left">
      Go Back
    </Button>
    <Button type="primary" icon="arrow-right" iconPosition="right">
      Go Forward
    </Button>
  </Button.Group>
  <Button.Group>
    <Button type="primary" icon="download" square={true} />
    <Button type="primary" icon="upload" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="success" icon="smile" square={true} />
    <Button type="success" icon="view" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="warning" icon="alipay" square={true} />
    <Button type="warning" icon="wxpay" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="danger" icon="delete" square={true} />
    <Button type="danger" icon="close" square={true} />
  </Button.Group>
</Playground>

## Props

<PropsTable of={Button} />
