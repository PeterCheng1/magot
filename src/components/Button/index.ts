export { default, ButtonProps, ButtonType } from './Button';
