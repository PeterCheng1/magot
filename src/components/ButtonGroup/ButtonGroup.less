@import '../../variable.less';

.mgt-button-group {
  .box-sizing();
  .box();

  position: relative;
  display: inline-block;

  & > .mgt-button {
    &:not(:first-child) {
      margin-left: -1px;
    }

    &:first-child {
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
    }

    &:last-child {
      border-top-left-radius: 0;
      border-bottom-left-radius: 0;
    }

    &:not(:first-child):not(:last-child) {
      border-radius: 0;
    }
  }

  & > .mgt-button-primary {
    .group-btn-divider(@btn-primary-border-color);
  }
  & > .mgt-button-success {
    .group-btn-divider(@btn-success-border-color);
  }
  & > .mgt-button-warning {
    .group-btn-divider(@btn-warning-border-color);
  }
  & > .mgt-button-danger {
    .group-btn-divider(@btn-danger-border-color);
  }
}

.group-btn-divider(@border) {
  &:not(:first-child) {
    border-left-color: lighten(@border, 12%);
  }
}
