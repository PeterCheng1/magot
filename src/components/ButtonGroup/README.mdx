---
name: ButtonGroup 按钮组
route: /components/ButtonGroup
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Button from '../Button';

# ButtonGroup 按钮组

按钮组合组件，用于组合一组功能相关的按钮。

## Usage

### 基础按钮组

<Playground>
  <Button.Group>
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
  <Button.Group>
    <Button type="primary" icon="arrow-left">
      Go Back
    </Button>
    <Button type="primary" icon="arrow-right" iconPosition="right">
      Go Forward
    </Button>
  </Button.Group>
  <Button.Group>
    <Button type="primary" icon="download" square={true} />
    <Button type="primary" icon="upload" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="success" icon="smile" square={true} />
    <Button type="success" icon="view" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="warning" icon="alipay" square={true} />
    <Button type="warning" icon="wxpay" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="danger" icon="delete" square={true} />
    <Button type="danger" icon="close" square={true} />
  </Button.Group>
</Playground>

### 按钮组尺寸

<Playground>
  <Button.Group size="small">
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
  <Button.Group>
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
  <Button.Group size="large">
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
</Playground>

### 不可用按钮组

<Playground>
  <Button.Group disabled={true}>
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
  <Button.Group disabled={true}>
    <Button type="warning" icon="alipay" square={true} />
    <Button type="warning" icon="wxpay" square={true} />
  </Button.Group>
</Playground>

## Props

<PropsTable of={Button.Group} />
