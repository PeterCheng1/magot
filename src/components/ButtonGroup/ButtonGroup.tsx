import React from 'react';

import './ButtonGroup.less';

import * as component from '../component';
import { ButtonProps } from '../Button/Button';

export interface ButtonGroupProps
  extends component.ComponentBase,
    component.DisableComponent,
    component.SizedComponent,
    component.NestedComponent {
  children: Array<React.ReactElement<ButtonProps>>;
}

const defaultProps: Partial<ButtonGroupProps> = {
  ...component.getDefaultDisabledProps(),
  ...component.getDefaultSizedProps(),
};

function ButtonGroup(props: ButtonGroupProps) {
  const cls = component.getComponentClasses('button-group', props);
  const btns = props.children.map((child, index) => {
    return React.cloneElement(child, {
      disabled: props.disabled,
      size: props.size,
      key: index,
    });
  });
  return (
    <div className={cls} style={props.style}>
      {btns}
    </div>
  );
}

ButtonGroup.defaultProps = defaultProps;

export default ButtonGroup;
