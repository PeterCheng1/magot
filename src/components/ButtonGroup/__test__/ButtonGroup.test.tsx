import React from 'react';
import { render } from 'enzyme';
import Button from '../../Button';
import ButtonGroup from '../ButtonGroup';
import { injectTestSuites } from '../../../../test/shared.test';

describe('ButtonGroup', () => {
  it('should renders dom correctly', () => {
    const wrapper = render(
      <ButtonGroup>
        <Button>L</Button>
        <Button>M</Button>
        <Button>R</Button>
      </ButtonGroup>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders empty children', () => {
    const wrapper = render(<ButtonGroup />);
    expect(wrapper.children()).toHaveLength(0);
  });

  it('should renders `size` prop when passed in', () => {
    const wrapper = render(
      <Button.Group size="small">
        <Button>L</Button>
        <Button>M</Button>
        <Button>R</Button>
      </Button.Group>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-button-sm')).toHaveLength(3);
  });

  it('should renders `disabled` prop when passed in', () => {
    const wrapper = render(
      <Button.Group disabled={true}>
        <Button>L</Button>
        <Button>M</Button>
        <Button>R</Button>
      </Button.Group>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-button-disabled')).toHaveLength(3);
  });

  injectTestSuites(Button.Group, { backgroundColor: '#f0f0f0' }, true);
});
