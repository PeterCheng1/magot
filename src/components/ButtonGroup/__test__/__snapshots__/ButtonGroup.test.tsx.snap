// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should renders \`disabled\` prop when passed in 1`] = `
<div
  class="mgt-button-group mgt-button-group-disabled"
>
  <button
    class="mgt-button mgt-button-disabled"
    disabled=""
    type="button"
  >
    L
  </button>
  <button
    class="mgt-button mgt-button-disabled"
    disabled=""
    type="button"
  >
    M
  </button>
  <button
    class="mgt-button mgt-button-disabled"
    disabled=""
    type="button"
  >
    R
  </button>
</div>
`;

exports[`ButtonGroup should renders \`size\` prop when passed in 1`] = `
<div
  class="mgt-button-group mgt-button-group-sm"
>
  <button
    class="mgt-button mgt-button-sm"
    type="button"
  >
    L
  </button>
  <button
    class="mgt-button mgt-button-sm"
    type="button"
  >
    M
  </button>
  <button
    class="mgt-button mgt-button-sm"
    type="button"
  >
    R
  </button>
</div>
`;

exports[`ButtonGroup should renders custom className 1`] = `
<div
  class="mgt-button-group jest-test-class"
/>
`;

exports[`ButtonGroup should renders custom styles 1`] = `
<div
  class="mgt-button-group"
  style="background-color:#f0f0f0"
/>
`;

exports[`ButtonGroup should renders dom correctly 1`] = `
<div
  class="mgt-button-group"
>
  <button
    class="mgt-button"
    type="button"
  >
    L
  </button>
  <button
    class="mgt-button"
    type="button"
  >
    M
  </button>
  <button
    class="mgt-button"
    type="button"
  >
    R
  </button>
</div>
`;
